lumen/sources/duckdb.py

```diff
@@ -134,6 +136,35 @@ def __init__(self, **params):
 
             self.tables = processed_tables
 
+        # Setup mirrors on first connection access
+        if self.mirrors:
+            _ = self._connection  # This will trigger connection creation and mirror setup
+
+    @property
+    def _connection(self):
+        """Thread-safe connection property - each thread gets its own connection to shared DB."""
+        if not hasattr(self._local, 'connection') or self._local.connection is None:
+            # Use shared memory database if uri is :memory:
+            # This allows multiple connections to share the same database
+            uri = self.uri
+            if uri == ':memory:':
+                uri = f':memory:shared_db_{id(self)}'
```

Comment:
I don't think this is appropriate. This means that two instances won't share the same DB, which is expected and required, e.g. when we call `create_sql_expr_source(...)`